flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -373,6 +376,16 @@ public void visit(RemoteMethodCallActionNode remoteMethodCallActionNode) {
         }
     }
 
+    private static String extractMcpToolKitName(String sourceCode) {
+        Pattern pattern = Pattern.compile("name:\\s*\"([^\"]*)\"");
+        Matcher matcher = pattern.matcher(sourceCode);
+        String nameValue = DEFAULT_MCP_SERVER_NAME;
+        if (matcher.find()) {
+            nameValue = matcher.group(1);
+        }
+        return nameValue;
+    }
+
```

Comment:
Instead of using regex patterns, we should traverse the syntax tree to derive the necessary information and gracefully handle the edge cases.

Reply:
Fixed in 894f82c